libraries/AP_HAL/DSP.cpp

```diff
@@ -28,23 +28,27 @@ extern const AP_HAL::HAL &hal;
 #define SQRT_2_3 0.816496580927726f
 #define SQRT_6   2.449489742783178f
 
-DSP::FFTWindowState::FFTWindowState(uint16_t window_size, uint16_t sample_rate)
+DSP::FFTWindowState::FFTWindowState(uint16_t window_size, uint16_t sample_rate, uint8_t harmonics)
     : _window_size(window_size),
     _bin_count(window_size / 2),
-    _bin_resolution((float)sample_rate / (float)window_size)
+    _bin_resolution((float)sample_rate / (float)window_size),
+    _harmonics(harmonics)
 {
     // includes DC ad Nyquist components and needs to be large enough for intermediate steps
     _freq_bins = (float*)hal.util->malloc_type(sizeof(float) * (window_size), DSP_MEM_REGION);
+    _derivative_freq_bins = (float*)hal.util->malloc_type(sizeof(float) * (_bin_count + 1), DSP_MEM_REGION);
     _hanning_window = (float*)hal.util->malloc_type(sizeof(float) * (window_size), DSP_MEM_REGION);
     // allocate workspace, including Nyquist component
     _rfft_data = (float*)hal.util->malloc_type(sizeof(float) * (_window_size + 2), DSP_MEM_REGION);
 
     if (_freq_bins == nullptr || _hanning_window == nullptr || _rfft_data == nullptr) {
```

Comment:
missing check for alloc fail _derivative_freq_bins

Reply:
Fixed